### PR TITLE
Support for shapes (Circle, Rounded rectangle)

### PR DESCRIPTION
**Pull request by adsamcik**
This pr adds #18
However to implement this I had to change quite a bit because most of the functions used radius. I wasn't sure what to leave  I wrote them in a way that allows anyone to easily add their own shape. Also the Rounded rectangle I wrote supports currently only API 21+ because I didn't have the time to backport it for older APIs.

### CODE REVIEW

File: spotlight/src/main/java/com/takusemba/spotlight/shapes/RoundedRectangle.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.takusemba.spotlight.shapes;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PointF;
+import android.support.annotation.RequiresApi;
+import android.view.View;
+
+@RequiresApi(21)
```
Comment (TakuSemba) [inline spotlight/src/main/java/com/takusemba/spotlight/shapes/RoundedRectangle.java]: i dont want this shape restricted to api level 21, isnt there any way to draw without this restriction?

File: spotlight/src/main/java/com/takusemba/spotlight/SimpleTarget.java [hunk 2]
```diff
@@ -11,147 +11,145 @@
...
-                                }
-                            });
-                    break;
-                case BELOW_SPOTLIGHT:
-                    layout.setY((int) (point.y + radius + 100));
-                    break;
-            }
-        }
-    }
+	private View view;
```
Comment (TakuSemba) [inline spotlight/src/main/java/com/takusemba/spotlight/SimpleTarget.java]: it seems like the whole indent had been changed. Im using `default` code style in Android Studio. Is it possible not to make these changes

File: spotlight/src/main/java/com/takusemba/spotlight/shapes/Shape.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.takusemba.spotlight.shapes;
+
+import android.animation.ValueAnimator;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PointF;
+import android.view.View;
+
+public abstract class Shape {
+	protected PointF mPoint = new PointF(0f, 0f);
```
Comment (TakuSemba) [inline spotlight/src/main/java/com/takusemba/spotlight/shapes/Shape.java]: sorry, I dont want to use m prefix for this repo.

**Comment by TakuSemba** [comment]
Im so sorry for the late response, I added some commend.

Again, thank you for the PR, I really like the idea to show different shapes,
I will merge this if you could fix the comment and conflicts. thanks!

File: spotlight/src/main/java/com/takusemba/spotlight/shapes/RoundedRectangle.java (hunk 1 above)
Comment (adsamcik) [reply spotlight/src/main/java/com/takusemba/spotlight/shapes/RoundedRectangle.java]: There is no native implementation before 21, so it needs to be made by other means for which I don't currently have the time.

**Comment by adsamcik** [comment]
Because it took so long I already rewrote the whole library to Kotlin with features I needed and during that I discovered some issues. The one that I can remember that you should probably fix is that you use constant integers (pixels) in calculatePosition method (more precisely in setPadding in that method), instead of dp.

**Comment by TakuSemba** [comment]
thank you for reporting issues.
I will fix those.

and so sorry for taking so much time, I will look at the changes and merge this.

**Pull request merged**